Libs/DICOM/Core/ctkDICOMScheduler.cpp

```diff
@@ -829,7 +829,9 @@ void ctkDICOMScheduler::stopJobsByJobUIDs(const QStringList &jobUIDs)
       job->setStatus(ctkAbstractJob::JobStatus::Stopped);
```

Comment:
We cannot set the status of the job here. It is not stopped yet. We can set a "request cancel" flag by callaing a job->requestCancel(), then let the job stop itself as soon as it can.

In general, calling a method of an object from another thread is not allowed. Methods that are specially hardened to be callable safely from another thread must be very clearly distinguished. A clean solution is to make all public method of an object thread-safe, but that is often very expensive to do (both in term of implementation and computation overhead). So, it may be better to distinguish methods that can be called from other threads by naming convention and documentation, for example these special methods can be called `request....()`.